codecov_cli/services/staticanalysis/analyzers/general.py

```diff
@@ -16,10 +17,26 @@ def _count_elements(self, node, types):
             count += 1
         return count
 
-    def _get_max_nested_conditional(self, node):
-        return (1 if node.type in self.condition_statements else 0) + max(
-            (self._get_max_nested_conditional(x) for x in node.children), default=0
-        )
+    def _get_max_nested_conditional(self, head):
+        """Iterates over all nodes in a function body and returns the max nested conditional depth.
+        Uses BFS to avoid recursion calls (so we don't throw RecursionError)
+        """
+        nodes_to_visit = deque()
+        nodes_to_visit.append([head, 0])
```

Comment:
i think you want
```
nodes_to_visit.append([head, 1 if head.type in self.condition_statements else 0])
```

thinking of:
```
if foo:
    pass
```
old code would return 1, new code would return 0 unless you change it

Reply:
Ah yes, you're right. Thanks :)